### ISSUE TITLE
DSP panic and no sound after wakeup - Dell 9700

### ISSUE THREAD

**Issue by maaarghk** [thesofproject/linux#2479]
<details>
<summary>dmesg of driver start (doesn't have full debug unfortunately)</summary>

```
Sep 27 13:07:19 applecross kernel: snd: loading out-of-tree module taints kernel.
Sep 27 13:07:19 applecross kernel: snd: module verification failed: signature and/or required key missing - tainting kernel
Sep 27 13:07:20 applecross kernel: sof-audio-pci 0000:00:1f.3: DSP detected with PCI class/subclass/prog-if info 0x040100
Sep 27 13:07:20 applecross kernel: sof-audio-pci 0000:00:1f.3: enabling device (0000 -> 0002)
Sep 27 13:07:20 applecross kernel: sof-audio-pci 0000:00:1f.3: DSP detected with PCI class/subclass/prog-if 0x040100
Sep 27 13:07:21 applecross kernel: sof-audio-pci 0000:00:1f.3: bound 0000:00:02.0 (ops i915_audio_component_bind_ops [i915])
Sep 27 13:07:21 applecross kernel: sof-audio-pci 0000:00:1f.3: use msi interrupt mode
Sep 27 13:07:21 applecross kernel: sof-audio-pci 0000:00:1f.3: hda codecs found, mask 4
Sep 27 13:07:21 applecross kernel: sof-audio-pci 0000:00:1f.3: Firmware info: version 1:5:1-88707
Sep 27 13:07:21 applecross kernel: sof-audio-pci 0000:00:1f.3: Firmware: ABI 3:16:0 Kernel ABI 3:17:0
Sep 27 13:07:21 applecross kernel: sof-audio-pci 0000:00:1f.3: warning: unknown ext header type 3 size 0x1c
Sep 27 13:07:21 applecross kernel: sof-audio-pci 0000:00:1f.3: warning: unknown ext header type 4 size 0x10
Sep 27 13:07:21 applecross kernel: sof-audio-pci 0000:00:1f.3: Topology: ABI 3:16:0 Kernel ABI 3:17:0
Sep 27 13:07:21 applecross kernel: sof-audio-pci 0000:00:1f.3: ASoC: Parent card not yet available, widget card binding deferred
Sep 27 13:07:21 applecross kernel: input: sof-soundwire Headset Jack as /devices/pci0000:00/0000:00:1f.3/sof_sdw/sound/card0/input27
Sep 27 13:07:21 applecross kernel: input: sof-soundwire HDMI/DP,pcm=5 as /devices/pci0000:00/0000:00:1f.3/sof_sdw/sound/card0/input28
Sep 27 13:07:21 applecross kernel: input: sof-soundwire HDMI/DP,pcm=6 as /devices/pci0000:00/0000:00:1f.3/sof_sdw/sound/card0/input29
Sep 27 13:07:21 applecross kernel: input: sof-soundwire HDMI/DP,pcm=7 as /devices/pci0000:00/0000:00:1f.3/sof_sdw/sound/card0/input30
```
</details>

<details>
<summary>dmesg when error occurs</summary>

```
Sep 29 19:45:21 applecross kernel: sof-audio-pci 0000:00:1f.3: error : DSP panic!
Sep 29 19:45:21 applecross kernel: sof-audio-pci 0000:00:1f.3: status: fw entered - code 00000005
Sep 29 19:45:21 applecross kernel: sof-audio-pci 0000:00:1f.3: error: can't enter idle
Sep 29 19:45:21 applecross kernel: sof-audio-pci 0000:00:1f.3: error: trace point 00004000
Sep 29 19:45:21 applecross kernel: sof-audio-pci 0000:00:1f.3: error: panic at src/lib/agent.c:62
Sep 29 19:45:21 applecross kernel: sof-audio-pci 0000:00:1f.3: error: DSP Firmware Oops
Sep 29 19:45:21 applecross kernel: sof-audio-pci 0000:00:1f.3: EXCCAUSE 0x0000003f EXCVADDR 0x00000000 PS       0x00060725 SAR     0x00000000
Sep 29 19:45:21 applecross kernel: sof-audio-pci 0000:00:1f.3: EPC1     0x00000000 EPC2     0xbe00d1fe EPC3     0x00000000 EPC4    0x00000000
Sep 29 19:45:21 applecross kernel: sof-audio-pci 0000:00:1f.3: EPC5     0x00000000 EPC6     0x00000000 EPC7     0x00000000 DEPC    0x00000000
Sep 29 19:45:21 applecross kernel: sof-audio-pci 0000:00:1f.3: EPS2     0x00060d20 EPS3     0x00000000 EPS4     0x00000000 EPS5    0x00000000
Sep 29 19:45:21 applecross kernel: sof-audio-pci 0000:00:1f.3: EPS6     0x00000000 EPS7     0x00000000 INTENABL 0x00000000 INTERRU 0x00000222
Sep 29 19:45:21 applecross kernel: sof-audio-pci 0000:00:1f.3: stack dump from 0xbe05a110
Sep 29 19:45:21 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a110: be05a140 00000001 be013760 00000001
Sep 29 19:45:21 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a114: 00000000 00000000 0000003e be065100
Sep 29 19:45:21 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a118: eea6b700 4bd5a5ca 000c0800 00000000
Sep 29 19:45:21 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a11c: 0dead000 00000000 3e82b018 ffffa3bd
Sep 29 19:45:21 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a120: c20c9504 ffffffff 3dd49440 ffffa3bd
Sep 29 19:45:21 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a124: 9dd08c25 ffffffff eea6b700 4bd5a5ca
Sep 29 19:45:21 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a128: 34622d80 ffffa3bd 8188fc48 ffffa96f
Sep 29 19:45:21 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a12c: 34622d80 ffffa3bd 00000000 00000000
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: error: ipc timed out for 0x80010000 size 216
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: status: fw entered - code 00000005
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: error: can't enter idle
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: error: trace point 00004000
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: error: panic at src/lib/agent.c:62
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: error: DSP Firmware Oops
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: EXCCAUSE 0x0000003f EXCVADDR 0x00000000 PS       0x00060725 SAR     0x00000000
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: EPC1     0x00000000 EPC2     0xbe00d1fe EPC3     0x00000000 EPC4    0x00000000
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: EPC5     0x00000000 EPC6     0x00000000 EPC7     0x00000000 DEPC    0x00000000
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: EPS2     0x00060d20 EPS3     0x00000000 EPS4     0x00000000 EPS5    0x00000000
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: EPS6     0x00000000 EPS7     0x00000000 INTENABL 0x00000000 INTERRU 0x00000222
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: stack dump from 0xbe05a110
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a110: be05a140 00000001 be013760 00000001
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a114: 00000000 00000000 0000003e be065100
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a118: 5e3b0b00 c1ab096f 4c8da558 ffffa3bd
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a11c: 00000000 00000000 c20b14a2 ffffffff
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a120: 83f9b8a8 ffffa96f 5e3b0b00 c1ab096f
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a124: c20a8045 ffffffff 0000000c 00000000
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a128: 4ac6ca18 ffffa3bd 49447280 ffffa3bd
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a12c: 4ac6ca20 ffffa3bd 334b4a00 ffffa3bd
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: error: hda irq intsts 0x00000000 intlctl 0xc0000080 rirb 00
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: error: dsp irq ppsts 0x00000000 adspis 0x00000000
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: error: host status 0x00000000 dsp status 0x00000000 mask 0x00000003
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: error: failed to set dai config for iDisp3
Sep 29 20:24:26 applecross kernel: sof-audio-pci 0000:00:1f.3: ASoC: error at snd_soc_dai_hw_params on iDisp3 Pin: -110
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: error: ipc timed out for 0x80010000 size 216
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: status: fw entered - code 00000005
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: error: can't enter idle
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: error: trace point 00004000
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: error: panic at src/lib/agent.c:62
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: error: DSP Firmware Oops
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: EXCCAUSE 0x0000003f EXCVADDR 0x00000000 PS       0x00060725 SAR     0x00000000
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: EPC1     0x00000000 EPC2     0xbe00d1fe EPC3     0x00000000 EPC4    0x00000000
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: EPC5     0x00000000 EPC6     0x00000000 EPC7     0x00000000 DEPC    0x00000000
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: EPS2     0x00060d20 EPS3     0x00000000 EPS4     0x00000000 EPS5    0x00000000
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: EPS6     0x00000000 EPS7     0x00000000 INTENABL 0x00000000 INTERRU 0x00000222
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: stack dump from 0xbe05a110
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a110: be05a140 00000001 be013760 00000001
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a114: 00000000 00000000 0000003e be065100
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a118: 5e3b0b00 c1ab096f 4c8da558 ffffa3bd
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a11c: 00000000 00000000 c20b14a2 ffffffff
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a120: 83f9b8a8 ffffa96f 5e3b0b00 c1ab096f
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a124: c20a8045 ffffffff 0000000c 00000000
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a128: 4ac6ca18 ffffa3bd 49447280 ffffa3bd
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a12c: 4ac6ca20 ffffa3bd 334b4a00 ffffa3bd
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: error: hda irq intsts 0x00000000 intlctl 0xc0000080 rirb 00
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: error: dsp irq ppsts 0x00000000 adspis 0x00000000
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: error: host status 0x00000000 dsp status 0x00000000 mask 0x00000003
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: error: failed to set dai config for iDisp3
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: ASoC: error at snd_soc_dai_hw_params on iDisp3 Pin: -110
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: error: ipc timed out for 0x80010000 size 216
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: status: fw entered - code 00000005
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: error: can't enter idle
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: error: trace point 00004000
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: error: panic at src/lib/agent.c:62
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: error: DSP Firmware Oops
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: EXCCAUSE 0x0000003f EXCVADDR 0x00000000 PS       0x00060725 SAR     0x00000000
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: EPC1     0x00000000 EPC2     0xbe00d1fe EPC3     0x00000000 EPC4    0x00000000
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: EPC5     0x00000000 EPC6     0x00000000 EPC7     0x00000000 DEPC    0x00000000
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: EPS2     0x00060d20 EPS3     0x00000000 EPS4     0x00000000 EPS5    0x00000000
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: EPS6     0x00000000 EPS7     0x00000000 INTENABL 0x00000000 INTERRU 0x00000222
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: stack dump from 0xbe05a110
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a110: be05a140 00000001 be013760 00000001
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a114: 00000000 00000000 0000003e be065100
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a118: 5e3b0b00 c1ab096f 4c8da558 ffffa3bd
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a11c: 00000000 00000000 c20b14a2 ffffffff
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a120: 83f9b8a8 ffffa96f 5e3b0b00 c1ab096f
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a124: c20a8045 ffffffff 0000000c 00000000
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a128: 4ac6ca18 ffffa3bd 49447280 ffffa3bd
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: 0xbe05a12c: 4ac6ca20 ffffa3bd 334b4a00 ffffa3bd
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: error: hda irq intsts 0x00000000 intlctl 0xc0000080 rirb 00
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: error: dsp irq ppsts 0x00000000 adspis 0x00000000
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: error: host status 0x00000000 dsp status 0x00000000 mask 0x00000003
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: error: failed to set dai config for iDisp3
Sep 29 20:24:27 applecross kernel: sof-audio-pci 0000:00:1f.3: ASoC: error at snd_soc_dai_hw_params on iDisp3 Pin: -110
```
</details>

This must be intermittent because this is not the first sleep/wakeup since boot. Running SOF driver as at commit https://github.com/thesofproject/linux/commit/779efb50ac86cd0e313ba3d56642eab9894e2e91

unloading and reloading the driver resulted in audio working again. 

Happy to try patches which add extra logging or anything like that and report back if it reoccurs...

**Comment by plbossart** [comment]
it's a known issue, version 1.6 of the firmware should solve this problem, can you try with binaries from 

https://github.com/thesofproject/sof-bin/tree/stable-v1.6

@lgirdwood FYI that makes 2 users of SOF in Scotland :-)

**Comment by maaarghk** [comment]
thanks, apologies I did not see anything when searching the issue tracker! seems 1.6 is also from today so I can hardly fault my distro for not having it also :-) 

before upgrading to 1.6 I am going to try to help resolve this chronic bikeshedding issue it has with the go.sh script as I see a release was made without one today.

also hi liam, I wouldn't say I'm any more of a user than your average laptop-haver though :) 


when I get 1.6 installed I will reboot and try to reach 1 week uptime without issues and then close the issue as assumed fixed if that suits you? (edit: 1.6 installed on my PC and a fix submitted to the sof-bin repo in the mean time!)

**Comment by maaarghk** [comment]
(sorry, not sure where to report this) I got some bad audio there on 1.6, sounded like a "bitcrush" effect. It went away after a moment and I can't see any log output although debug isn't on. Is there something I can do to help debug that?

**Comment by plbossart** [comment]
this was supposed to be a major release, that's not good if the first user reports a problem - Murphy's Law applies I guess.... 

Can you provide more details on what happened, and if you can reproduce your problem. providing a pointer to the alsa-info information would help.

@lgirdwood FYI, we'll probably need to help debug this

**Comment by maaarghk** [comment]
Here's alsa-info: http://alsa-project.org/db/?f=8a504cb7dde33bfb232e07ae7b59739ce1632b65

I was not using the laptop at the time, but I got a system notification (slack noise) which sounded bitcrushed so I then played a youtube video in Chrome to check and got the same effect. I stopped sound on that and waited a moment and the next thing I played was normal again.

Fingers crossed for you that I simply forgot to upgrade some other critical component :)

**Comment by plbossart** [comment]
Unfortunately that seems the same issue as before, the PANIC used to happen when transient sounds were played. Now the panic can't happen any longer but I guess the firmware still goes in the weeds, albeit in a recoverable way.  

That's not as bad as the previous PANIC which required a reboot, but that's still not a passing grade to me.

**Comment by maaarghk** [comment]
obviously I'm not sure but I would say I have been using the 1.5.1 firmware on this hardware for a few months and it rarely would have the panic, and only after a long uptime, whereas this happened very soon after a reboot.

can you remind me how to enable debug output in dmesg and I will enable that and keep an ear out for anything else?

**Comment by maaarghk** [comment]
ok, it is ongoing at the moment (i think it just started with this youtube video so I don't think anything to do with a transient sound, I didn't notice one anyway). it hasn't gone away as I keep playing this video so maybe something that happens when the output is silent clears this issue. i could be wrong but I believe it is only from the left speaker.

short video: https://photos.app.goo.gl/jTZDGzPGSP8QckDf7

alsa-info during issue: http://alsa-project.org/db/?f=63f36924c956b96a8c06975505719ffecb55e903


edit: Tried to record with crecord but seems I'll need to recompile my driver with DEBUG_PROBES config to do that - but I am sort of in the dark and could be going in the totally wrong direction.

I noticed one other thing  - the distortion volume increases and decreases with the output volume, it is not constant. I don't really know if that means anything but it seems confusing from an outside perspective when the only thing I think I updated is firmware. I took a video showing this and also demo'ed the problem isn't with chrome at least, as the system sound generated when changing volume is also affected (Although hard to hear on the video I suppose) - https://photos.app.goo.gl/TyoTGeJuxsYAumtf9 

edit2: another check, I plugged in headphones and changed output to headphones and they did not exhibit the distortion, but then changed back and the speakers still exhibit the distortion (unless sufficient silent time is passed, I don't have a high enough rate of this happening to measure that!)

edit3: the relevant outputs in alsamixer (distortion goes away when muted) are `rt1308-1 DAC L` and `rt1308-1 DAC R` - if you're unfamiliar with the hardware, as a reminder, these are both for the left channel despite the name but one seems to be doing a woofer and another a tweeter.

**Comment by plbossart** [comment]
dynamic debug can be enabled by copying this file 
[ddebug.txt](https://github.com/thesofproject/linux/files/5302503/ddebug.txt) as /etc/modprobe.d/sof-ddebug.conf

**Comment by maaarghk** [comment]
https://gist.github.com/maaarghk/602109b21a679b649786357bae3c1309

It seems whenever the pcm stream is closed then re-opened it has a 50/50 chance of reopening with the issue being exhibited. I just have notification sounds going off just now every few minutes and some are fine and some are not. they correlate with "sof_sdw sof_sdw: ASoC: find BE for widget Low Latency Playback 2" lines.

not sure if there is enough info here?

**Comment by paulstelian97** [comment]
For the devs on your platform to know where to try to look at first, this sounds like a frame goes really in the weeds (frame sync coming up incorrectly). DAI and codec configuration can be problematic.

**Comment by plbossart** [comment]
Oh man, this is a SoundWire-based device.  I thought it was the usual HDAudio platform but no Sir, it's brand-new hardware for which we completed the upstream process literally 10 days ago.

One hypothesis is that your kernel does not have all the latest and greatest patches for SoundWire. I don't know what Manjaro does.

Is it possible for you to install a kernel based on our kernel topic/sof-dev branch to give it a try? You'd have exactly the same configuration as what we test internally at Intel.

**Comment by plbossart** [comment]
@bardliao do you have a CML-H device? Wondering if we can try to reproduce a test case with short notifications,

**Comment by maaarghk** [comment]
Hi Pierre, I will try to compile and run the whole kernel from topic/sof-dev branch. I already made a dkms tree (https://github.com/maaarghk/soundwire-dkms/blob/8b578ac2e0a1126b14fc43a09e981e0cded811f0/dkms.conf) to build a lot of modules from https://github.com/thesofproject/linux/commit/779efb50ac86cd0e313ba3d56642eab9894e2e91 but of course I may have missed something.

**Comment by plbossart** [comment]
If this helps, we have a set of typical kconfigs at https://github.com/thesofproject/kconfig
https://github.com/thesofproject/kconfig/blob/master/kconfig-sof-default.sh should give you a config that compiles in about 4mn and gives you what you need (I use it on a CML-U device)

**Comment by ruffson** [comment]
I am running Fedora 32 with a `5.8.11-200` kernel on a Precision 5750 (same hardware).
If there is an easy enough way for me to test it, I can check too.

**Comment by maaarghk** [comment]
If all sound is currently working well for you, you can clone https://github.com/thesofproject/sof-bin/tree/stable-v1.6 (checkout branch stable-v1.6) and run go.sh to install the 1.6 firmware. on rebooting you should have bad audio sometimes like on my video. 

I'm now running 5.9.0-rc7 compiled from topic/sof-dev and I just had the problem again with distortion, identical issue. On this occasion both times I was playing music and there were not any notification sounds.

**Comment by ruffson** [comment]
> If all sound is currently working well for you ...

No I did not touch this on my current install because I did not want to mess around and rather wait.

**Comment by plbossart** [comment]
@ruffson and @maaarghk can you provide the results of 'cat /sys/class/dmi/id/product_sku" - this will help check which version of SoundWire devices you have (the sku number is used in our machine drivers in sound/soc/intel/boards/sof_sdw.c). Thanks!

**Comment by ruffson** [comment]
> @ruffson and @maaarghk can you provide the results of 'cat /sys/class/dmi/id/product_sku" - this will help check which version of SoundWire devices you have (the sku number is used in our machine drivers in sound/soc/intel/boards/sof_sdw.c). Thanks!

Sure:

```
$ cat /sys/class/dmi/id/product_sku
0990
```

**Comment by plbossart** [comment]
> > @ruffson and @maaarghk can you provide the results of 'cat /sys/class/dmi/id/product_sku" - this will help check which version of SoundWire devices you have (the sku number is used in our machine drivers in sound/soc/intel/boards/sof_sdw.c). Thanks!
> 
> Sure:
> 
> ```
> $ cat /sys/class/dmi/id/product_sku
> 0990
> ```

ok, that's a 2-amp 4-speaker system, the more complicated one. https://github.com/thesofproject/linux/blob/96b9d7cf347158efac3d830d9937e8efc39579c8/sound/soc/intel/boards/sof_sdw.c#L93

**Comment by maaarghk** [comment]
I have 098f which is the same driver_data

**Comment by plbossart** [comment]
@ruffson @maaarghk making sure you see the other report on Dell Precision 5750 at https://github.com/thesofproject/sof/issues/3495

**Comment by maaarghk** [comment]
@ruffson I don't know if you still feel like you would rather wait but you can consider installing dkms module I made following steps on my [blog](https://blog.fts.scot/2020/07/04/dell-xps-2020-how-to-get-audio-working-on-linux/) but I wouldn't know how to help you with that on fedora specifically.

**Comment by ruffson** [comment]
> @ruffson I don't know if you still feel like you would rather wait but you can consider installing dkms module I made following steps on my [blog](https://blog.fts.scot/2020/07/04/dell-xps-2020-how-to-get-audio-working-on-linux/) but I wouldn't know how to help you with that on fedora specifically.

I don't wanna that guy who wants other people to fix his problems without helping out at all, but I rely on this system right now and do not want to mess with it too much. I am sorry. (Esp. if there is no guarantee that this setup even works with Fedora in the same way). Also, your blog seems to be down.

**Comment by bardliao** [comment]
> @bardliao do you have a CML-H device? Wondering if we can try to reproduce a test case with short notifications,

Yes, I can reproduce the issue on my CML-H device. The reproducing rate is about 10% on my side and I don't find a way to easily reproduce it yet.

**Comment by plbossart** [comment]
I suggest we all comment on https://github.com/thesofproject/sof/issues/3499 so let's close this one.
